src/mWater_collate/mWater_pull_collate.Rmd

```diff
@@ -0,0 +1,166 @@
+---
+title: "mWater_pull_collate"
+author: "Sam Struthers"
+date: "`r Sys.Date()`"
+output: html_document
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+library(tidyverse)
+
+#grab site metadata
+sampling_sites <- read_csv("data/water_sampling_sites.csv")
+# sort for
+upper_sites <- sampling_sites%>%
+  filter(watershed != "CLP  Mainstem-Fort Collins")%>%
+  mutate(site_code = tolower(site_code))
+```
+
+# API Pull and download raw sensor notes
+
+```{r}
+# 
+creds = yaml::read_yaml("src/mWater_collate/mWater_API.yml")
+api_url = as.character(creds["url"])
+rm(creds)
+#create raw download spreadsheet
+download.file(api_url, destfile = "data/mWater_raw_sensor_notes.csv")
+
+```
+
+
+# Tidying
+
+## ALL NOTES
+```{r}
+#read in file
+all_notes <- read_csv(file = "data/mWater_raw_sensor_notes.csv")%>%
+  mutate(
+    #start dt comes in UTC -> to MST
+    start_dt = with_tz(start_dt, tz = "America/Denver"),
+         date = as.Date(start_dt),
+         time_start = format(start_dt, "%H:%M"),
+    # If other is chose, make site == other response and make it lowercase     
+    site = ifelse(site == "Other (please specify)", tolower(site_other), site),
+    site = ifelse(site %in% upper_sites$site_code, toupper(site), site),
+         site_other = NULL,
+         #this part does not work yet
+         # (cant get rid of (please specify))
+         visit_type = ifelse(str_detect(visit_type, "Other"),
+                             str_replace(visit_type, "Other (please specify),", visit_type_other),
+                             visit_type),
+         #If other is chosen, make photos downloaded equal to response
+         photos_downloaded = ifelse(photos_downloaded == "Other (please specify)", photos_downloaded_other, photos_downloaded),
+         photos_downloaded_other = NULL, 
+         #Date/field season columns to match QAQC workflow
+         DT_round = floor_date(start_dt, "15 minutes"),
+         DT_join = as.character(DT_round),
+         field_season = year(DT_round),
+         last_site_visit = DT_round) %>%
+#arrange by most recent visit
+    arrange(DT_round)
+```
+
+
+## Sensor Notes
+
+```{r}
+#grab only notes where technician is interacting with sensor
+sensor_notes <- all_notes%>%
+  filter(grepl("Sensor",visit_type))%>%
+  select(-c(chla_volume_ml, vol_filtered_blank_dup, sample_collected,do_mgl,
+            cond_ms_cm, temp_c, upstream_pic, downstream_pic, clarity, other_pic, other_pic_descriptor))%>%
+  mutate(sonde_employed = case_when(is.na(sensor_change)  ~ 1,
+                                    sensor_change == "Pulled" ~ 0,
+                                    sensor_change %in% c("Swapped", "Deployed") ~ 1), 
+         # Create basis for calibration report name
+         # this will be used to check for calibration report in data files and then b
+         cal_report_name = case_when(cal_report_collected == TRUE ~ paste0(site, "_", format(start_dt, "%Y%m%d")),
```

Comment:
question: what is the log stuff doing? Does mWater have a way to download and interact with the calibration report stuff?

Reply:
Not yet! I want to build out a way to make sure that techs/us are uploading all our files in a timely manner. This will essentially be the file that R will look for in the data folder to see if the file has been uploaded or not